src/v/security/audit/schemas/schemas.h

```diff
@@ -176,5 +182,49 @@ class ocsf_base_event : public ocsf_base_impl {
 
         return h;
     }
+
+    /// Method to estimate the size of an ocsf message
+    ///
+    /// This works by iterating on the fields within the tuple returned by the
+    /// 'equality_fields()' method. This method is implemented on all ocsf
+    /// structures.
+    ///
+    /// The returned size is 'estimated' because it doesn't take into account
+    /// things like padding and any fields that a struct contains that it did
+    /// not include in its return value for 'equality_fields()'. Currently
+    /// the only example of this is the 'network_endpoint' struct.
+    template<typename T>
+    size_t estimated_ocsf_size(const T& t) const noexcept {
+        size_t sz = 0;
+        if constexpr (reflection::is_std_vector<T>) {
+            for (const auto& element : t) {
+                sz += estimated_ocsf_size(element);
+            }
+        } else if constexpr (reflection::is_std_optional<T>) {
+            sz += 1;
+            if (t.has_value()) {
+                sz += estimated_ocsf_size(*t);
+            }
+        } else if constexpr (std::is_same_v<ss::sstring, T>) {
+            sz += t.size();
```

Comment:
how precise do you want to be? because this will be off by 40 or so bytes if the string is long enough so you'd want `t.size() + sizeof(ss::sstring)`